### PR TITLE
Feature/auth endpoints meta

### LINKED ISSUE #239: Auth refactoring: meta app
parent issue - #224

* [x] update all bff endpoints with a new permission mixin
* [x] update all the tests

### PR DESCRIPTION
**Pull request by annashamray**
fixes #239
depends on #244

### CODE REVIEW

File: backend/src/zac/core/api/views.py [hunk 1]
```diff
@@ -492,7 +492,7 @@ def get_queryset(self) -> List[dict]:
         return zaaktypen
 
     def get_zaaktypen(self) -> List[dict]:
-        zaaktypen = get_zaaktypen(UserPermissions(self.request.user))
+        zaaktypen = get_zaaktypen(self.request.user)
```
Comment (damm89) [inline backend/src/zac/core/api/views.py:495]: Line 32: from zac.accounts.permissions import UserPermissions

Is unused.

File: backend/src/zac/core/blueprints.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 from elasticsearch_dsl.query import Q, Query, Range, Term
 from rest_framework import serializers
+from zgw_consumers.api_models.catalogi import ZaakType
```
Comment (damm89) [inline backend/src/zac/core/blueprints.py:5]: Unused?

Just like line 3: import Q

File: backend/src/zac/core/views/zaken.py [hunk 3]
```diff
@@ -75,7 +75,7 @@ class Index(PermissionRequiredMixin, BaseListView):
 
     def get_filter_form_kwargs(self):
         kwargs = super().get_filter_form_kwargs()
-        kwargs["zaaktypen"] = get_zaaktypen(UserPermissions(self.request.user))
+        kwargs["zaaktypen"] = get_zaaktypen(self.request.user)
```
Comment (damm89) [inline backend/src/zac/core/views/zaken.py:78]: Line 14 
from zac.accounts.permissions import UserPermissions
is now unused

File: backend/src/zac/elasticsearch/drf_api/views.py [hunk 4]
```diff
@@ -63,12 +63,11 @@ def post(self, request, *args, **kwargs):
         return Response(zaak_serializer.data)
 
     def perform_search(self, data) -> List[Zaak]:
-        user_perms = UserPermissions(self.request.user)
         # TODO search on zaaktype attributes instead of urls
         if data.get("zaaktype"):
             zaaktype_data = data.pop("zaaktype")
             zaaktypen = get_zaaktypen(
-                user_perms,
+                self.request.user,
```
Comment (damm89) [inline backend/src/zac/elasticsearch/drf_api/views.py:70]: Same comment as above:

line 11
from zac.accounts.permissions import UserPermissions
is now unused

File: backend/src/zac/reports/views.py [hunk 5]
```diff
@@ -23,7 +23,7 @@ def get_queryset(self):
         qs = super().get_queryset()
 
         # filter on allowed zaaktypen
-        zaaktypen = get_zaaktypen(UserPermissions(self.request.user))
+        zaaktypen = get_zaaktypen(self.request.user)
```
Comment (damm89) [inline backend/src/zac/reports/views.py:26]: line 11
from zac.accounts.permissions import UserPermissions

is now unused

**Review by damm89 — CHANGES_REQUESTED** [review]

**Review by sergei-maertens — APPROVED** [review]
Agree with the remarks from Daniël :)

File: backend/src/zac/core/api/views.py (hunk 1 above)
Comment (annashamray) [reply backend/src/zac/core/api/views.py:495]: unused import is removed

File: backend/src/zac/core/blueprints.py (hunk 2 above)
Comment (annashamray) [reply backend/src/zac/core/blueprints.py:5]: removed, thanks!

File: backend/src/zac/core/views/zaken.py (hunk 3 above)
Comment (annashamray) [reply backend/src/zac/core/views/zaken.py:78]: removed!

File: backend/src/zac/elasticsearch/drf_api/views.py (hunk 4 above)
Comment (annashamray) [reply backend/src/zac/elasticsearch/drf_api/views.py:70]: removed

File: backend/src/zac/reports/views.py (hunk 5 above)
Comment (annashamray) [reply backend/src/zac/reports/views.py:26]: also removed

**Review by annashamray — COMMENTED** [review]
@damm89 Thank you very much for your comments. The major code cleanning in the #255 - I do my best to remove all unused code and imports there

**Pull request merged**